phoenix-core/src/main/java/org/apache/phoenix/coprocessor/ScanRegionObserver.java

```diff
@@ -218,7 +228,11 @@ protected RegionScanner doPostScannerOpen(final ObserverContext<RegionCoprocesso
         if (j != null) {
             innerScanner = new HashJoinRegionScanner(innerScanner, p, j, tenantId, c.getEnvironment());
         }
-
+        if (scanOffset != null) {
+            innerScanner = getOffsetScanner(c, innerScanner,
+                    new OffsetResultIterator(new RegionScannerResultIterator(innerScanner), scanOffset),
+                    scan.getAttribute(QueryConstants.LAST_SCAN) == null ? false : true);
```

Comment:
Nit: change this:

    scan.getAttribute(QueryConstants.LAST_SCAN) == null ? false : true

to this:

    scan.getAttribute(QueryConstants.LAST_SCAN) != null